eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Providers/OperationMethodProviders/PageableOperationMethodProvider.cs

```diff
@@ -102,9 +119,9 @@ protected MethodBodyStatement[] BuildBodyStatements()
             arguments.AddRange(_contextualPath.PopulateArguments(This.As<ArmResource>().Id(), requestMethod.Signature.Parameters, contextVariable, _signature.Parameters));
 
             // Handle ResourceData type conversion if needed
-            if (IsResourceDataType(_itemType))
+            if (_resourceClient != null)
```

Comment:
consider to add come comment that `_resourceClient != null` means _itemType is ResourceData, or rename `_resourceClient` to be more meaningful